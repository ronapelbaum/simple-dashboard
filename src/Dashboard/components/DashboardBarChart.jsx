import * as React from 'react';
import {Paper} from '@material-ui/core';
import {
  Chart,
  BarSeries,
  Title,
  ArgumentAxis,
  ValueAxis,
} from '@devexpress/dx-react-chart-material-ui';

import { Animation } from '@devexpress/dx-react-chart';
import { connect } from 'react-redux';
import { loadingModel, impressionsModel, impressionsBarDef } from '../model';

const DashboardBarChart = ({ barDef, data, loading}) => {
  if (!data || loading) {
    return null;
  }
  return (
    <Paper>
      <Chart data={data}>
        <Title text={barDef.title} />
        <ArgumentAxis />
        <ValueAxis />
        <BarSeries
          valueField={barDef.valueField}
          argumentField={barDef.argumentField}
        />
        <Animation />
      </Chart>
    </Paper>
  );
};

const mapStateToProps = (state) => ({
  loading: loadingModel(state),
  data: impressionsModel(state),
  barDef: impressionsBarDef(),
});

export default connect(mapStateToProps)(DashboardBarChart);
