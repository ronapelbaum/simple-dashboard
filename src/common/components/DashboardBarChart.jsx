import * as React from 'react';
import {
  Chart,
  BarSeries,
  Title,
  ArgumentAxis,
  ValueAxis,
} from '@devexpress/dx-react-chart-material-ui';
import { Animation } from '@devexpress/dx-react-chart';

const DashboardBarChart = ({ config, data, loading}) => {
  if (!data || loading) {
    return null;
  }
  return (
    <Chart data={data}>
      <Title text={config.title} />
      <ArgumentAxis />
      <ValueAxis />
      <BarSeries
        valueField={config.valueField}
        argumentField={config.argumentField}
      />
      <Animation />
    </Chart>
  );
};

export default DashboardBarChart;
