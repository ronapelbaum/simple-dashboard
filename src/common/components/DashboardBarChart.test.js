import React from "react";
import { shallow } from "enzyme";
import DashboardBarChart from './DashboardBarChart';

describe("DashboardBarChart.jsx suite", () => {
  it("renders correctly for empty", () => {
    expect(shallow(<DashboardBarChart />).debug()).toMatchSnapshot();
  });
  it("renders correctly", () => {
    const data = [
      { key: 'a', value: '1' },
      { key: 'a', value: '2' },
    ];
    const config = {
      title: "bar chart",
      valueField: 'value',
      argumentField: 'key',
    };
    expect(shallow(<DashboardBarChart data={data} config={config} />).debug()).toMatchSnapshot();
  });
});
