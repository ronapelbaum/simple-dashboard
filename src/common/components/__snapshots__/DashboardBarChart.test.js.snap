// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardBarChart.jsx suite renders correctly 1`] = `
"<Component data={{...}}>
  <Component text=\\"bar chart\\" />
  <Component />
  <Component />
  <Component valueField=\\"value\\" argumentField=\\"key\\" barWidth={0.9} />
  <AnimationBase easing={[Function: easeOutCubic]} duration={1000} />
</Component>"
`;

exports[`DashboardBarChart.jsx suite renders correctly for empty 1`] = `""`;
